containous/maesh
# Use custom env and file loaders to load Maesh config

## What does this PR do?

This PR:

- Refactors some code to be consistent and more explicit.
- Adds a custom env loader to load config from env vars with a `MAESH_` prefix.
- Adds a custom file loader to load config from a file with a maesh name.

Fixes #539.

### How to test it

* Build binary
* Run Maesh: `MAESH_LOGLEVEL=info ./dist/maesh`, config should be loaded from env var.
* Create a config file named `maesh.yaml` in the current dir.
* Run Maesh: `./dist/maesh`, config should be loaded from the file.

### Additional Notes

The `env` and `file` configuration loaders for mesh proxies remains unchanged. Therefore, config is loaded from env vars prefixed by `TRAEFIK_` and from a file resolved as described [here](https://docs.traefik.io/getting-started/configuration-overview/#the-static-configuration).
